### PR TITLE
feat: [RTD-492] Limit PUT PGP files on APIM

### PR DESCRIPTION
**Pull request by TommasoLencioni**
This PR proposes to impose a limit on PUT request body size of upload API.

### List of changes
- modify policy to check whether Content-Length is between 0 and a configurable cap (currently 500MB).

### Motivation and context
With this change we can define an allowed body size and reject payloads that exceeds the threshold.
The limit is customizable with a variable.

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [x] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

```
-target=module.api_azureblob
```
---

### CODE REVIEW

**Review by petretiandrea — APPROVED** [review]
🥳

File: src/core/api/azureblob/azureblob_authorizative_policy.xml [hunk 1]
```diff
@@ -26,6 +26,33 @@
         </when>
       </choose>
 
+    <set-variable name="allowedSize" value="${pgp-put-limit-bytes}" />
+    <set-variable name="bodySize" value="@(context.Request.Headers["Content-Length"][0])" />
+
+    <!-- Check for Content-Length not zero-->
+    <choose>
```
Comment (GiovanniMancini) [inline src/core/api/azureblob/azureblob_authorizative_policy.xml:33]: @TommasoLencioni Please let's check content length header we are actually receiving from senders

**Review by GiovanniMancini — CHANGES_REQUESTED** [review]
We should check Content Length as set by actual senders

File: src/core/api/azureblob/azureblob_authorizative_policy.xml (hunk 1 above)
Comment (TommasoLencioni) [reply src/core/api/azureblob/azureblob_authorizative_policy.xml:33]: @GiovanniMancini checked, we receive Content-Length header from each sender currently onboard 👍

**Pull request merged**